cpp/src/io/utilities/kvikio_manager.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/io/kvikio_manager.hpp>
+#include <cudf/logger.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <kvikio/defaults.hpp>
+
+#include <cuda_runtime.h>
+
+#include <mutex>
+
+namespace cudf::io {
+
+namespace detail {
+
+kvikio_initializer::kvikio_initializer()
+{
+  static std::once_flag flag{};
+  std::call_once(flag, [] {
+    // Call before any cuFile API calls to ensure the CUDA context is initialized.
+    // Workaround for https://github.com/rapidsai/cudf/issues/14140, where cuFileDriverOpen errors
+    // out if no CUDA calls have been made before it. This is a no-op if the CUDA context is already
+    // initialized.
+    cudaFree(nullptr);
```

Comment:
This initialization should happen in kvikio rather than cudf, imo. This seems like an important behavior for all kvikio users and not just cudf, right?

Also, I don't like using an initializer class as a data member. The previous design where an idempotent function was called, like `kvikio_integration::set_up_kvikio();`, is a better design, because there is no state involved and the data member is never accessed. It should be thought of more like a scoped lock, where you create an object and let RAII handle things _in a local scope_. Is there a problem with doing that?